src/shared/gitlab-client.ts

```diff
@@ -149,4 +159,90 @@ export class GitlabClient {
     };
   }
 
+  public async getUserByEmail(email: string) {
+    if (!validateEmail(email)) {
+      throw Boom.badRequest(`Invalid email '${email}'`);
+    }
+    const search = {
+      search: email,
+    };
+    const users = await this.fetchJson<User[]>(`users?${qs.stringify(search)}`, true);
+    if (!users || !users.length) {
+      throw Boom.badRequest(`Can\'t find user '${email}'`);
+    }
+    if (users.length > 1) {
+      // This shoud never happen
+      const message = `Found multiple users with email '${email}'`;
+      this.logger.warning(message);
+      throw Boom.badRequest(message);
+    }
+    return users[0];
+  }
+
+  public createUser(
+    email: string,
+    password: string,
+    username: string,
+    name: string,
+    externUid?: string,
+    provider?: string,
+    confirm = false,
+  ) {
+    const newUser = {
+      email,
+      password,
+      username,
+      name,
+      extern_uid: externUid,
+      provider,
+      confirm,
+    };
+    return this.fetchJson<User>(`users`, {
+      method: 'POST',
+      body: JSON.stringify(newUser),
+      headers: {
+        'content-type': 'application/json',
+      },
+    }, true);
+  }
+
+  public addUserToGroup(userId: number, teamId: number, accessLevel = 30) {
```

Comment:
It's the default, meaning *developer access*. Here's more about them: https://docs.gitlab.com/ee/api/members.html. Still, it's quite unclear what they allow or disallow.